ports/raspberrypi/supervisor/internal_flash.c

```diff
@@ -51,25 +51,36 @@
 #define NO_CACHE 0xffffffff
 STATIC uint8_t _cache[SECTOR_SIZE];
 STATIC uint32_t _cache_lba = NO_CACHE;
+STATIC uint32_t _flash_size = 0;
 
 void supervisor_flash_init(void) {
     bi_decl_if_func_used(bi_block_device(
         BINARY_INFO_MAKE_TAG('C', 'P'),
         "CircuitPython",
         RESERVED_FLASH,
-        TOTAL_FLASH_SIZE - RESERVED_FLASH,
+        (1 * 1024 * 1024), // This is a minimum. We can't set it dynamically.
         NULL,
         BINARY_INFO_BLOCK_DEV_FLAG_READ |
         BINARY_INFO_BLOCK_DEV_FLAG_WRITE |
         BINARY_INFO_BLOCK_DEV_FLAG_PT_UNKNOWN));
+
+    // Read the RDID register to get the flash capacity.
+    uint8_t cmd[] = {0x9f, 0, 0, 0};
+    uint8_t data[4];
+    flash_do_cmd(cmd, data, 4);
+    uint8_t power_of_two = 21;
+    if (data[3] >= 20 && data[3] < 30) {
+        power_of_two = data[3];
+    }
```

Comment:
Are there cases where `data[3]` is out of range (too small, probably)? It looks like you just assume 2MB (2**21)? Could you add a comment explaing this